converters/toZigbee.js

```diff
@@ -1030,23 +1031,34 @@ const converters = {
         convertSet: async (entity, key, value, meta) => {
             if (key == 'color') {
                 const result = await converters.light_color.convertSet(entity, key, value, meta);
-                if (result.state && result.state.color.hasOwnProperty('x') && result.state.color.hasOwnProperty('y')) {
-                    result.state.color_temp = Math.round(libColor.ColorXY.fromObject(result.state.color).toMireds());
-                }
-
                 return result;
             } else if (key == 'color_temp' || key == 'color_temp_percent') {
                 const result = await converters.light_colortemp.convertSet(entity, key, value, meta);
-                result.state.color = libColor.ColorXY.fromMireds(result.state.color_temp).rounded(4).toObject();
                 return result;
             }
         },
         convertGet: async (entity, key, meta) => {
-            if (key == 'color') {
-                await converters.light_color.convertGet(entity, key, meta);
-            } else if (key == 'color_temp') {
-                await converters.light_colortemp.convertGet(entity, key, meta);
+            /**
```

Comment:
@Koenkk this one is nearly the same as light_color, but it also reads colorTemperature.

Maybe we should love the building of `attributes` array to a separate function in lib/light.js, we can then call it in both light_color.convertGet and light_color_colortemp.convertGet. Something like

```function readLightColorAttributes(meta, additionalAttributes=[]) {}```

Then we can just set additionalAttributes to `['colorTemperature']` for light_color_colortemp.

Reply:
Agree